### PR TITLE
refactor(ProjectService): ComponentContent interface and Component class

### LINKED ISSUE #872: ComponentContent interface and Component class
- Define a ComponentContent interface that represents the fields that all components have.
   - Add nodeId: string to this object
- Change ProjectService.getComponent() to return a ComponentContent type instead of any.
- Change ComponentHeader to take in ComponentContent, and remove nodeId input. Get it from ComponentContent instead.

This will make the code clearer.

### PR DESCRIPTION
**Pull request by hirokiterashima**
## Changes
- Define ComponentContent interface and child interfaces (e.g. MatchContent, GraphContent, etc)
   - ProjectService.getComponent() now returns ComponentContent 
   - ProjectService.getComponents() now returns ComponentContent[]
- Define Component class that has ComponentContent and nodeId fields
   - Instantiate Component in StudentComponent and pass this into ComponentHeader, replacing ```any``` and ```nodeId```
- Change code and templates to work with the new classes

## Test
- VLE, AT, CM work as before, in particular: 
   - Prompt and dynamic prompt

Closes #872

### CODE REVIEW

File: src/app/services/projectService.spec.ts [hunk 1]
```diff
@@ -232,14 +233,15 @@ function shouldGetTheComponent() {
...
-    const componentExists2 = service.getComponent('node9', 'mnzx68ix8h');
+    const componentExists2 = service.getComponent('node9', 'mnzx68ix8h') as EmbeddedContent;
     expect(componentExists2).not.toBe(null);
     expect(componentExists2.type).toEqual('embedded');
     expect(componentExists2.url).toEqual('NewtonScooters-potential-kinetic.html');
   });
 }
 
 function shouldGetTheComponentsByNodeId() {
+  ``;
```
Comment (geoffreykwan) [inline src/app/services/projectService.spec.ts]: Remove line that was accidentally left in.

File: src/assets/wise5/components/multipleChoice/edit-multiple-choice-connected-components/edit-multiple-choice-connected-components.component.spec.ts [hunk 2]
```diff
@@ -6,6 +6,8 @@ import { ProjectService } from '../../../services/projectService';
 import { EditMultipleChoiceConnectedComponentsComponent } from './edit-multiple-choice-connected-components.component';
 import { createConnectedComponentObject } from '../../../../../app/authoring-tool/edit-connected-components/edit-connected-components.component.spec';
 import { StudentTeacherCommonServicesModule } from '../../../../../app/student-teacher-common-services.module';
+import { ComponentContent } from '../../../common/ComponentContent';
```
Comment (geoffreykwan) [inline src/assets/wise5/components/multipleChoice/edit-multiple-choice-connected-components/edit-multiple-choice-connected-components.component.spec.ts]: Remove unneeded import line.

**Review by geoffreykwan — CHANGES_REQUESTED** [review]
Works well. It's a little confusing now that getComponent() returns a ComponentContent and is often set to a variable called component and now in component-student.component.ts there's a component variable that's a Component. These naming inconsistencies can be resolved later down the line though.

I found some lines of code that can be removed.

**Comment by hirokiterashima** [comment]
Yes, let's rename the variables in a future task.

**Review by geoffreykwan — APPROVED** [review]
Looks good.

**Review by breity — APPROVED** [review]
Looks good.

**Pull request merged**